src/coreclr/debug/createdump/dumpwriterelf.h

```diff
@@ -0,0 +1,67 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#ifdef HOST_64BIT
+#define ELF_CLASS ELFCLASS64
+#else
+#define ELF_CLASS ELFCLASS32
+#endif
+
+#define Ehdr   ElfW(Ehdr)
+#define Phdr   ElfW(Phdr)
+#define Shdr   ElfW(Shdr)
+#define Nhdr   ElfW(Nhdr)
+#define auxv_t ElfW(auxv_t)
+
+#if defined(__x86_64__)
+#define ELF_ARCH  EM_X86_64
+#elif defined(__i386__)
+#define ELF_ARCH  EM_386
+#elif defined(__aarch64__)
+#define ELF_ARCH  EM_AARCH64
+#elif defined(__arm__)
+#define ELF_ARCH  EM_ARM
+#endif
+
+#define PH_HDR_CANARY 0xFFFF
+
+#ifndef NT_FILE
+#define NT_FILE		0x46494c45
+#endif
+
+class DumpWriter
```

Comment:
Should these derive from a base class to ensure the contract between createdump and the writers?

Reply:
I don't think it gives us any advantages since Linux vs MacOS is all conditionally compiled.  The "contract" surface is small enough (like 4 functions) that having them dup'ed in dumpwriterelf.h/dumpwritermacho.h isn't a big deal.  I already do something similar with CrashInfo.+